### PR TITLE
feat(canvas):Add multiple routes control

### PR DESCRIPTION
**Pull request by mmelko**
Added controls for the multiple-routes support including setting visible route.

### CODE REVIEW

File: packages/ui/src/components/Visualization/ContextToolbar/ContextToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { FunctionComponent, useContext } from 'react';
+import { Chip, ToolbarItem } from '@patternfly/react-core';
+
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/ContextToolbar/ContextToolbar.tsx:3]:
```suggestion
```

File: packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                //  validator={ValidationService.validateUniqueName}
+                onClick={() => {
+                  onSelectFlow(flow.id);
+                }}
+                onChange={(name) => {
+                  setFlowName(flow.id, name);
+                }}
+              />
+              //TODO add description
+              <p>{''}</p>
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.tsx:74]: Maybe this is not necessary, as it shows on the UI
image



```suggestion
```

File: packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        </Tr>
+      </Thead>
+      <Tbody>
+        {flows.map((flow: BaseVisualCamelEntity) => (
+          <Tr key={flow.id} data-testid={`flows-list-row-${flow.id}`}>
+            <Td dataLabel={columnNames.current.id}>
+              <InlineEdit
+                data-testid={`goto-btn-${flow.id}`}
+                value={flow.id}
+                //  validator={ValidationService.validateUniqueName}
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.tsx:65]: [curious-question] Can we bring this functionality over?

File: packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+const onSelect = jest.fn();
+const renderWithContext = () => {
+  return render(
+    <EntitiesContext.Provider value={contextValue as unknown as EntitiesContextResult}>
+      <FlowTypeSelector onSelect={onSelect} />
+    </EntitiesContext.Provider>,
+  );
+};
+describe('FlowTypeSelector.tsx', () => {
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx:32]:
```suggestion

describe('FlowTypeSelector.tsx', () => {
```

File: packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  test('component renders', () => {
+    const wrapper = renderWithContext();
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx:34]: [:nail_care:] What do you think about keeping this abstraction over rendering the component but as a custom component instead.

The usage would be:

```typescript
render(<FlowTypeSelectorWithContext />)
```

And the component would be:
```jsx
const FlowTypeSelectorWithContext: FunctionComponent = () => {
  return (
    <EntitiesContext.Provider value={contextValue as unknown as EntitiesContextResult}>
        <FlowTypeSelector onSelect={onSelect} />
    </EntitiesContext.Provider>
  );
};

File: packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.test.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import { FlowsList } from './FlowsList';
+import { act, fireEvent, render } from '@testing-library/react';
+import { SourceSchemaType } from '../../../../models/camel';
+import { CamelRouteVisualEntity } from '../../../../models/visualization/flows';
+import { EntitiesContext } from '../../../../providers/entities.provider';
+import { EntitiesContextResult } from '../../../../hooks';
+import { VisualFlowsApi } from '../../../../models/visualization/flows/flows-visibility';
+
+//jest.mock(visibleFlov)
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.test.tsx:9]:
```suggestion
```

**Review by lordrip — CHANGES_REQUESTED** [review]
Thanks for bringing this @mmelko.

There are a few rough edges that we might need to polish (please see the related comments)

In addition to that, there might be missing functionality for `KameletBinding` and `Pipe` because the list doesn't work with them.

Also, it looks like the counter needs a refresh, because after adding a new route in the source code editor, notice how the counter is bigger than expected.

image

File: packages/ui/src/models/camel/source-schema-config.ts [hunk 7]
```diff
@@ -48,6 +54,10 @@ class SourceSchemaConfig {
       this.config[type].schema = schema;
     }
   }
+
+  getAsList(): ISourceSchema[] {
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/ui/src/models/camel/source-schema-config.ts:58]: I was planning to use that but then I switch to using  SourceSchemaType IRRC. So I can delete this and use in the test `Object.values()`

File: packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx (hunk 5 above)
Comment (lordrip) [reply packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx:34]: Yes, mostly readability.

File: packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+  return (
+    <Select
+      id="flows-list-select"
+      isOpen={isOpen}
+      onOpenChange={(isOpen) => {
+        setIsOpen(isOpen);
+      }}
+      toggle={toggle}
+      // minWidth="400px"
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsMenu.tsx:59]: attribute doesn't exist anymore

**Comment by mmelko** [comment]
@lordrip please check :)

**Review by lordrip — APPROVED** [review]

**Pull request merged**